### PR TITLE
robot_name, v1

### PR DESCRIPTION
**Pull request by bap13**
More practice with Ruby style.

### CODE REVIEW

File: robot_name/robot_name.rb [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+  class << self
+    attr_accessor :registry
+  end
+  @registry ||= []
 
   def initialize(args = {})
-    @@registry ||= []
-    @name_generator = args[:name_generator]
+    name_generator = args.fetch(:name_generator, false)
+    @name = name_generator ? name_generator.call : generate_new_name
```
Comment (jmmastey) [inline robot_name/robot_name.rb]: Use `method(:generate_new_name)` and you can get a callable reference in either case. That'll clear up these few lines.

File: robot_name/robot_name.rb [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
+  private
 
-    if @name_generator
-      @name = @name_generator.call
-    else
-      generate_char = -> { ('A'..'Z').to_a.sample }
-      generate_num = -> { rand(10) }
+  def generate_new_name
+    generate_char = -> { ('A'..'Z').to_a.sample }
+    generate_num = -> { rand(10) }
```
Comment (jmmastey) [inline robot_name/robot_name.rb]: If we're doing short methods, these really are begging to be moved out into methods.

File: robot_name/robot_name.rb [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
 
-      @name = "#{generate_char.call}#{generate_char.call}#{generate_num.call}#{generate_num.call}#{generate_num.call}"
-    end
+    [Array.new(2) { generate_char.call },
+     Array.new(3) { generate_num.call }].join
```
Comment (jmmastey) [inline robot_name/robot_name.rb]: Ehhhh, not sure I like this form of Array initialization. It's clever, but harder to understand overall. Prefer clarity to brevity.

**Comment by jmmastey** [comment]
A few awkward constructions still, but overall looks fine. Nice work.

File: robot_name/robot_name.rb (hunk 1 above)
Comment (bap13) [inline robot_name/robot_name.rb]: Nice! I didn't know you could do that.

**Comment by bap13** [comment]
Thanks for your comments Joe!

**Pull request closed without merging**